src/google-maps/map-marker/map-marker.ts

```diff
@@ -380,64 +391,42 @@ export class MapMarker implements OnInit, OnDestroy, MapAnchorPoint {
     return this.marker;
   }
 
-  private _combineOptions(): Observable<google.maps.MarkerOptions> {
-    return combineLatest([this._options, this._title, this._position, this._label, this._clickable])
-        .pipe(map(([options, title, position, label, clickable]) => {
-          const combinedOptions: google.maps.MarkerOptions = {
-            ...options,
-            title: title || options.title,
-            position: position || options.position,
-            label: label || options.label,
-            clickable: clickable !== undefined ? clickable : options.clickable,
-            map: this._googleMap.googleMap,
-          };
-          return combinedOptions;
-        }));
+  /** Creates a combined options object using the passed-in options and the individual inputs. */
+  private _combineOptions(): google.maps.MarkerOptions {
+    const options = this._options || DEFAULT_MARKER_OPTIONS;
+    return {
+      ...options,
+      title: this._title || options.title,
+      position: this._position || options.position,
+      label: this._label || options.label,
+      clickable: this._clickable !== undefined ? this._clickable : options.clickable,
+      map: this._googleMap.googleMap,
+    };
   }
 
-  private _watchForOptionsChanges() {
-    this._options.pipe(takeUntil(this._destroy)).subscribe(options => {
-      if (this.marker) {
-        this._assertInitialized();
-        this.marker.setOptions(options);
-      }
-    });
-  }
+  /** Syncs the input values with the marker object. */
+  private _syncInputs() {
+    const {marker, _title, _position, _label, _clickable} = this;
 
-  private _watchForTitleChanges() {
-    this._title.pipe(takeUntil(this._destroy)).subscribe(title => {
-      if (this.marker && title !== undefined) {
-        this._assertInitialized();
-        this.marker.setTitle(title);
+    if (marker) {
+      marker.setOptions(this._combineOptions());
```

Comment:
I think we need to use the "changes" object in ngOnChanges to prevent these methods from running if their particular input has not changed. The reason is that we don't necessarily want to re-set all of the options when we're changing one particular input. Imagine a situation where the developer sets an initial set of options for a marker. Then the user drags the marker across the map. Then they trigger a change to the title. This way of doing it will set all of the initial inputs that have not been updated yet, including the initial position, causing the marker to return to its original position.

If we use the changes object to see what input is actually being changed, we can just run that setter method and avoid this problem (similar to how I originally watched for various changes instead of running setOptions for every change).

Reply:
I was thinking about that, but my assumption was that the Google Maps API would know not to update the map if the values haven't changed.

Reply:
I believe that's correct, but in the situation I outlined, it would appear that the values had changed to the internal Marker class. Basically, if we initialize a marker with these options:
```
  [position]="{lat: 23, lng: 10}"
  [title]="'new marker'"
  [draggable]="true"
```
Since draggable is true, users can drag the marker anywhere they want. This will update the position on the internal marker, but it won't mutate the object that the develop input. So, let's say they drag it to {lat: 16, lng: -2}.

Then, let's say a user clicks on something that triggers an update to the title. This code will call setOptions with
```
{
  position: {lat: 23, lng: 10},
  title: 'updated title',
  draggable: true,
}
```
This snaps the marker out of the spot the user dragged it to, even though we would it expect to only update the title.

By relying on the "changes" object, we can check what input is actually being changed and then just run the appropriate setter, in this case "setTitle" which will update the title and not change the position.

Reply:
That makes sense. Changing...